LanguageFeatures/Augmentation-libraries/augmenting_constructors_A18_t04_lib.dart

```diff
@@ -20,3 +20,14 @@ augment class C {
   augment C.foo(int x): this.foo(x); // Cyclic! But it is fixed below
   augment C.foo(int x): this(x + 1); // Not a cyclic now
 }
+
+augment enum E {
+  augment e0;
+  augment const E.foo(int x): this.foo(x);
+  augment const E.foo(int x): this(x + 1);
```

Comment:
I think this should be an error, but https://github.com/dart-lang/language/pull/4157 will clarify that.